report_thesis/src/sections/proposed_approach/optimization_framework.tex

```diff
@@ -0,0 +1,193 @@
+
+The role of the objective function is to provide the metric data that the sampler and pruner need to make informed decisions.
+As we step through each oxide and model in Lines~\ref{step:oxide_loop} to~\ref{step:model_loop}, we call the $objective$ function with the $Optimize\_Combined\_Objective$ function, seen in Line~\ref{step:optimize_combined_objective}.
+The $Optimize\_Combined\_Objective$ is returning the metrics for each trial, which the objective function then uses to guide the optimization process.
+
+This leads us to the \nameref{alg:combined_objective} function which, as previously mentioned, contains the optimization logic.
+It defines the systematic process of hyperparameter tuning, model training, and evaluation across multiple trials.
+
+To this function we supply the current model $M$, the number of trials to run $N$, the sampler, and the pruner, seen in Line~\ref{step:combined_objective_params}.
+The number of trials determines how many times the optimization process will be run for each model and oxide.
+This is an important parameter, as it regulates how much time the framework has to optimize a given model.
+In an ideal scenario, this number would be very high to ensure that the optimization process has identified the best possible configuration.
+However, depending on the number of models in consideration the number of trials can quickly become computationally prohibitive.
+
+In Lines~\ref{step:sample_hyperparameters} to~\ref{step:instantiate_dim_reduction} we instantiate the model and the preprocessors with the hyperparameters sampled by the sampler.
+The sampling process of these is what is being guided by the objective function based on the metrics returned.
+In our setup, we always require that a scaler is instantiated, to ensure that the data is correctly scaled, however, data transformation and dimensionality reduction are not always beneficial, therefore these can be set to the identity function.
+
+Once the preprocessors are instantiated, we construct a pipeline of these to ensure that the data is processed in the order these are defined, seen in Line~\ref{step:construct_pipeline}.
+The order is important, as the preprocessing steps are to a degree interdependent: scaling ensures that all features are on a similar range, which is ensures that transformations are applied uniformly.
+Similarly, dimensionality reduction techniques typically also produce better results if the data has been scaled.
+However, dimensionality reduction may not always be beneficial or may not always produce better results if the data has been transformed.
+As such, we allow for the optimization framework to optionally use these if they are beneficial.
+
+In Line~\ref{step:get_data} to Line~\ref{step:apply_pipeline_cv} we fetch the data, apply our data partitioning strategy to generate four folds, a training set and a test set, and apply the preprocessing to the folds and the training set.
```

Comment:
This is part of what I keep saying to fix. We _also_ apply it to the test set, but we don't fit the preprocessor to it.

![image](https://github.com/chhoumann/thesis-chemcam/assets/29108628/905b323c-8e64-4c68-90de-c4e2fcf2f311)


```suggestion
In Line~\ref{step:get_data} to Line~\ref{step:apply_pipeline_cv} we fetch the data, apply our data partitioning strategy to generate four cross-validation sets, a training set and a test set, and apply the preprocessing to the datasets.
```